workos/resources/sso.py

```diff
@@ -1,87 +1,59 @@
-from workos.resources.base import WorkOSBaseResource
+from typing import List, Literal, Union
 
+from workos.resources.workos_model import WorkOSModel
+from workos.typing.literals import LiteralOrUntyped
+from workos.utils.connection_types import ConnectionType
 
-class WorkOSProfile(WorkOSBaseResource):
-    """Representation of a User Profile as returned by WorkOS through the SSO feature.
 
-    Attributes:
-        OBJECT_FIELDS (list): List of fields a WorkOSProfile is comprised of.
-    """
+class Profile(WorkOSModel):
+    """Representation of a User Profile as returned by WorkOS through the SSO feature."""
 
-    OBJECT_FIELDS = [
-        "id",
-        "email",
-        "first_name",
-        "last_name",
-        "groups",
-        "organization_id",
-        "connection_id",
-        "connection_type",
-        "idp_id",
-        "raw_attributes",
-    ]
+    object: Literal["profile"]
+    id: str
+    connection_id: str
+    connection_type: LiteralOrUntyped[ConnectionType]
+    organization_id: Union[str, None]
```

Comment:
I had no idea it was possible to have a `Profile` or `Connection` with no `organization_id`

Reply:
Oh. OAuth. That's how.